eval/vars.go

```diff
@@ -0,0 +1,37 @@
+package eval
+
+import (
+	"context"
+	"maps"
+
+	"github.com/blackstork-io/fabric/parser/definitions"
+	"github.com/blackstork-io/fabric/pkg/ctyencoder"
+	"github.com/blackstork-io/fabric/pkg/diagnostics"
+	"github.com/blackstork-io/fabric/plugin"
+)
+
+// Evaluates `variables` and stores the results in `dataCtx` under the key "vars".
+func ApplyVars(ctx context.Context, variables *definitions.ParsedVars, dataCtx plugin.MapData) (diags diagnostics.Diag) {
+	if variables.Empty() {
+		return
+	}
+	var vars plugin.MapData
+
+	varsData := dataCtx["vars"]
```

Comment:
Maybe we should use const for `"vars"`?

Reply:
Unfortunantly all maps [are actually pointers](https://groups.google.com/g/golang-nuts/c/SjuhSYDITm4/m/jnrp7rRxDQAJ), even more of a pointer type than a slice, and so [you can't even create a const map](https://go.dev/play/p/ZPv7glrXOT8)